src/views/ajax.js

```diff
@@ -5,6 +5,36 @@ const arrowIcon = require('./open').arrowIcon;
 const icons = require('../assets/icons').default;
 const statusIcon = require('../assets/icons').statusIcon;
 
+/**
+ * removeOrigin - function to remove origin from a URL.
+ *
+ * @param {String} url URL to remove the origin from.
+ * @param {String} origin Origin to remove from URL.
+ */
+const removeOrigin = (url = '', origin = window.location.origin) => {
+  if (url.substr(0, origin.length) === origin) {
+      url = url.substr(origin.length, url.length);
+  }
+
+  return url;
+};
+
+/**
+ * removeOriginFromUrl - function to remove origin from a URL
+ *                       reguarding `http` and `https`.
+ *
+ * @param {String} url URL to remove the origin from.
+ * @param {String} origin Origin to remove from URL.
+ */
+const removeOriginFromUrl = (url = '', origin = window.location.origin) => {
+  url = url.trim();
+
+  url = removeOrigin(url, origin.replace(/^http\:\/\//, 'https://'));
```

Comment:
I think @legomushroom is probably better, but we would need to show the lock icon in that case. As for hud... if we don't have the lock icon then we need to keep both the scheme and the host in the cases where just the scheme is different (otherwise it would look funny), if we have the lock then we can strip the scheme and host in cases where the host/port are the same and always show the lock for https requests.